Apps/Src/DebugDump.c

```diff
@@ -0,0 +1,141 @@
+#include "os.h"
+#include "Tasks.h"
+#include "bsp.h"
+#include "CANbus.h"
+#include "CAN_Queue.h"
+#include "Pedals.h"
+#include "Minions.h"
+#include "Contactors.h"
+#include "common.h"
+#include <errno.h> 
+#include "Tasks.h"
+#include "SendTritium.h"
+
+// global variables
+extern fault_bitmap_t FaultBitmap;
+extern os_error_loc_t OSErrLocBitmap;
+extern bool UpdateVel_ToggleCruise;
+extern uint16_t SupplementalVoltage;
+extern uint32_t StateOfCharge;
+
+static const char *MINIONPIN_STRING[] = {
+    FOREACH_MinionPin(GENERATE_STRING)
+};
+
+static const char *CONTACTOR_STRING[] = {
+    FOREACH_contactor(GENERATE_STRING)
+};
+
+// Need to keep this in sync with Task.h
+/*----------------------------------------------*/
+#define FAULT_BITMAP_NUM 6
+#define OS_LOC_NUM 14
+static const char *OS_LOC_STRING[] = { 
+    "OS_NONE_LOC", 
+    "OS_ARRAY_LOC", 
+    "OS_READ_CAN_LOC", 
+    "OS_READ_TRITIUM_LOC",
+    "OS_SEND_CAN_LOC",
+    "OS_SEND_TRITIUM_LOC",
+    "OS_UPDATE_VEL_LOC",
+    "OS_CONTACTOR_LOC",
+    "OS_MINIONS_LOC",
+    "OS_MAIN_LOC",
+    "OS_CANDRIVER_LOC",
+    "OS_MOTOR_CONNECTION_LOC",
+    "OS_DISPLAY_LOC"
+};
+
+static const char *FAULT_BITMAP_STRING[] = { 
+    "FAULT_NONE", 
+    "FAULT_OS", 
+    "FAULT_UNREACH", 
+    "FAULT_TRITIUM",
+    "FAULT_READBPS",
+    "FAULT_DISPLAY",
+    "FAULT_BPS"
+};
+/*----------------------------------------------*/
+
+void Task_DebugDump(void* p_arg) {
+    OS_ERR err;
+    Minion_Error_t mErr;
+
+    while(1){
+
+        // Get pedal information
+        int8_t accelPedal = Pedals_Read(ACCELERATOR);
+        printf("ACCELERATOR: %d\n\r", accelPedal);
+
+        int8_t brakePedal = Pedals_Read(BRAKE);
+        printf("BRAKE: %d\n\r", brakePedal);
+
+        // Get minion information
+        for(MinionPin_t pin = 0; pin < MINIONPIN_NUM-1; pin++){ // Plan to change MINIONPIN_NUM-1 -> MINIONPIN_NUM after Minion function is fixed
```

Comment:
What does this mean?

Reply:
I was thinking of changing the minion read function to include outputs (the brakelight). But I feel this doesn't matter right now. This should happen when we move over to the embedded sharepoint. I feel like we should just leave the comment in there for now so we can be reminded when we port the code.